src/librustc_resolve/lib.rs

```diff
@@ -3614,6 +3606,12 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             }
         }
     }
+
+    fn report_privacy_errors(&self) {
+        for (span, string) in self.privacy_errors.iter() {
```

Comment:
Since `privacy_errors` is a `HashMap`, this will result in nondeterministic error ordering. Maybe the map should be changed to an `FnvHashMap` or a `BTreeMap`?